PIPELINE.md

```diff
@@ -0,0 +1,126 @@
+# Jenkins Pipeline
+
+## Overview
+
+The Jenkinsfile in this repository contains a sample pipeline that performs a
+build and applies object configurations to instantiate this application.
+
+The pipeline can be used to maintain a running application on an OpenShift
+cluster.
+
+## Basic: Getting Started
+
+1. Start an OpenShift cluster (origin v1.3 and later). Create a new project.
+2. [Optional] Create a fork of this repository
+3. Instantiate the pipeline from template by invoking:
+   ```
+   oc new-app -f https://raw.githubusercontent.com/FORK/parksmap-web/add-jenkinsfile/ose3/pipeline-buildconfig-template.json -p GIT_URI=https://github.com/FORK/parksmap-web.git
+   ```
+
+   (Where FORK is your own fork of this repository or `openshift-roadshow`)
+4. Start the pipeline build from the CLI with:
+   ```
+   oc start-build parksmap-web-pipeline
+   ```
+
+   OR
+
+   Start the pipeline from the web console. Navigate to the project, Builds->Pipelines.
+   Click `Start` next to the pipeline.
+
+
+## Advanced: Example of a PR-based development flow
+
+The Jenkinsfile in this repository can be used to support a PR flow with Jenkins on OpenShift.
+The main application is built and deployed to an 'integration' project. Every time a PR is
+submitted to the source repository, a new OpenShift project is created to build and test the
+PR and allow for manual testing. Once the manual testing is approved, the build succeeds.
+
+The following steps guide you through the setup of the Github Organization Folder plugin in
+Jenkins and the required steps on the GitHub side. The Github Organization Folder plugin
+instantiates a pipeline for every project it finds with a Jenkinsfile at its root. It also
+sets up webhooks in GitHub so that new PRs and changes to a branch are automatically built
+in Jenkins. In order for webhooks to work, the ngrok application is used to create a public
+endpoint for the internal Jenkins service.
+
+### GitHub Setup
+
+1. Create your own forks of [nationalparks](https://github.com/openshift-roadshow/nationalparks.git),
+   [mlbparks](https://github.com/openshift-roadshow/mlbparks.git), and
+   [parksmap-web](https://github.com/openshift-roadshow/parksmap-web.git).
+2. Rename each of those repositories in GitHub with a common prefix, for example:
```

Comment:
The reason is to make it easier to point to your repos in the Github Organization folder settings. However, I see that I missed saying that here. It's also not absolutely necessary, given that you can come up with a regular expression that includes all your project names. I'll update to remove this requirement and explain how you can specify your project names in the Github Organization folder settings.